### DISCUSSION TITLE
How to always install latest version of a specific package

### DISCUSSION THREAD

**Discussion by tabrez96** [pnpm/pnpm#5654] [Q&A]
We recently moved from yarn to pnpm. We have a private npm package, which is being used in an app. While installing packages in that app, it always installs the version of that private package from pnpm lock. I have to manually bump the version in package.json everytime to install latest version of that package. As a workaround, while working with yarn, we had a custom script to remove that package from yarn.lock, so that the version was not frozen and the latest version was installed always. Doing the same with pnpm gives this warning:
<img width="353" alt="Screenshot 2022-11-18 at 11 32 48 AM" src="https://user-images.githubusercontent.com/29143166/202632244-548dfca1-e439-40ef-a088-41a6ae2d9970.png">

What is the right way to prevent a dependency from being included in lock file?

**Comment by lvqq** [comment]
Try use `hooks.afterAllResolved`: https://pnpm.io/pnpmfile

**Reply by tabrez96** [reply]
Tried 'afterAllResolved', it is still giving the same error. This is my .pnpmfile.cjs:

```
function afterAllResolved(lockfile) {
  const myPackages = Object.keys(lockfile.packages).filter((packageName) =>
    packageName.includes('@company/package-name')
  );

  myPackages.forEach((packageName) => {
    /**
     * Delete packages from pnpm lock file
     */
    delete lockfile.packages[packageName];
  });

  return lockfile;
}

module.exports = {
  hooks: {
    afterAllResolved,
  },
};

```

**Comment by zkochan** [comment]
I guess the only way would be to add a prepare script to the project's package.json, which would run an update:

```json
{
  "scripts": {
    "prepare": "pnpm up -L pkg-name"
  }
}
```

**Reply by tabrez96** [reply]
Thanks! It does the job, but since it updates the package.json file with the new version and pnpm-lock file, we will end up having commits to update the version for every release of the package. Wondering if there is anyway to ignore this. Ideally, we don't want that package to be included in the lock file at all.
